python/cudf/cudf/tests/test_dataframe_copy.py

```diff
@@ -159,8 +160,8 @@ def test_kernel_deep_copy():
     gdf = DataFrame.from_pandas(pdf)
     cdf = gdf.copy(deep=True)
     sr = gdf["b"]
-
-    add_one[1, len(sr)](sr._column.data_array_view(mode="write"))
+    with CUDFNumbaConfig():
+        add_one[1, len(sr)](sr._column.data_array_view(mode="write"))
```

Comment:
This test surprises me. I expected that `CUDFNumbaConfig` would be treated as an implementation detail and thus not needed in our tests. (Ideally it should be `_CUDFNumbaConfig` to make it private.)

Do we _need_ to be testing a raw `@cuda.jit` function with reference to the private `_column` member? Is testing `apply` with a UDF the same thing, or does it matter that we are testing a raw column with a raw Numba JIT kernel?

Reply:
These seem like pretty old tests that are responsible for making sure shallow/deep copies behave as expected when some kind of action is taken that would modify the data inplace, in this case it looks like a numba kernel was chosen. I think I can sneak in a PR to update these.

Reply:
This was resolved separately in https://github.com/rapidsai/cudf/pull/13385